packages/ui-kit/src/components/TimeRangePicker/TimeRangePicker.types.ts

```diff
@@ -0,0 +1,51 @@
+import type { TimeRangeInput } from '../../helpers'
+import type { DefaultThemes } from '../ThemeProvider'
+
+export type DateRangeOptionsProps = {
+  /** A unique identifier for the date range option, used to distinguish between different options. */
+  uid: string
+
+  /** A human-readable label that describes the date range option. */
+  label?: string
+
+  /** The value of the date range option, which can be either a specific date range or a relative time range. */
+  value?: TimeRangeInput
+}
+
+export interface TimeRangePickerProps extends Omit<React.ComponentPropsWithoutRef<'div'>, 'onChange' | 'defaultValue'> {
+  /**
+   * The initial theme used as a base. It provides a default set of styling
+   * from which customizations can be applied.
+   */
+  baseTheme?: DefaultThemes
+
+  /** If set to true, disables the option within the picker that allows users to select a range extending to the current date. */
+  disableDateUntilNow?: boolean
+
+  /** If set to true, disables the option for users to select a custom date range. */
+  disableCustomRange?: boolean
+
+  /** If set to true, disables the option for selecting relative date ranges, such as "last n days." */
+  disableCustomRelative?: boolean
+
+  /** If set to true, disables all selectable options in the component, effectively making it read-only. */
+  disableOptions?: boolean
+
+  /** If true, the entire component is disabled, and users cannot interact with it. */
+  disabled?: boolean
```

Comment:
Nit: are `disableDateUntilNow`, `disableCustomRange`, etc., actually causing us to "omit" options from the picker? If so, maybe we should say `omitDateUntilNow`, `omitCustomRange`?

I'm just thinking, "disable" also means something in the context of web forms, and when I see `disableOptions` and `disabled`, I am wondering if these are omitting things or disabling visible options. Are there really two behaviors here?

Reply:
I've never seen the `omit` prefix used in this way in UI component libraries. More commonly, the `disable` prefix is used, for example, `disableElevation` in MUI's [Button](https://mui.com/material-ui/react-button/#contained-button) or various props with the `disable` prefix in [Autocomplete](https://mui.com/material-ui/api/autocomplete/). From my understanding, the main difference between `disableSomething` and `disabled` is that `disabled` refers to a state, whereas `disableSomething` is used to deactivate a specific feature.